std/os/file.zig

```diff
@@ -337,16 +337,16 @@ pub const File = struct {
                 }
 
                 assert(pos >= 0);
-                return math.cast(usize, pos);
+                return math.cast(u64, pos);
             },
             else => @compileError("unsupported OS"),
         }
     }
 
-    pub fn getEndPos(self: File) GetSeekPosError!usize {
+    pub fn getEndPos(self: File) GetSeekPosError!u64 {
         if (is_posix) {
             const stat = try os.posixFStat(self.handle);
-            return @intCast(usize, stat.size);
+            return @intCast(u64, stat.size);
```

Comment:
I think it's probably incorrect that the `size` field of all the `Stat` structs is signed. If we changed those to be unsigned, this `@intCast` could go away.

Reply:
Hmm, posix dictates that `st_size` is the size of the file in bytes, but that's for files only, everything else is fair game.

Don't know what the best solution would be here, let's return an error?

Changing the `Stat` structure is a no-go for me, the structure layout is dictated by the kernel and should not be changed.

Reply:
> Changing the `Stat` structure is a no-go for me, the structure layout is dictated by the kernel and should not be changed.

That's reasonable, and I agree, let's leave it signed in the struct. But just to clarify: whether the integer is signed or unsigned does not affect the layout/ABI. It's the same size and twos complement either way.

Relevant: https://stackoverflow.com/questions/12275831/why-is-the-st-size-field-in-struct-stat-signed

I think we have 2 choices here:
 * assert that the size can never be negative
 * return `error.Unexpected` if the size is negative. I see that `Unexpected` is already one of the possible errors in GetSeekPosError. If this ever was found to happen in real world software, we would introduce an actual error code for it. Unexpected means that the zig code wasn't aware that a particular error was possible.

I'll leave the decision up to you.